test/unittests/UserTest.php

```diff
@@ -151,27 +151,48 @@ class UserTest extends TestCase
         ]
     ];
 
-    private $_permInfo     = [0 => ['permID' => 1,
-        'code'        => "superuser",
-        'description' => "superuser description",
-        'categoryID'  => 1
-    ],
+    private $_permInfo = [
+        0 => ['permID' => 1,
+            'code'        => "superuser",
+            'description' => "superuser description",
+            'categoryID'  => 1,
+            'action'      => null,
+            'moduleID'    => null
+        ],
         1 => ['permID' => 2,
             'code'        => "test_permission",
             'description' => "description 1",
-            'categoryID'  => 2
+            'categoryID'  => 2,
+            'action'      => 'View',
+            'moduleID'    => 2
         ],
         2 => ['permID' => 3,
             'code'        => "test_permission2",
             'description' => "description 2",
-            'categoryID'  => 3
+            'categoryID'  => 3,
+            'action'      => 'Edit',
+            'moduleID'    => 5
         ],
         3 => ['permID' => 4,
             'code'        => "test_permission3",
             'description' => "description 3",
-            'categoryID'  => 4
+            'categoryID'  => 4,
+            'action'      => 'View/Create',
+            'moduleID'    => 5
         ]
     ];
+
```

Comment:
@driusan @kongtiaowang 

I had to modify the test class to have the PR pass checks, the problem is, because I need toinstantiate a module to get its Long name, i resorted to use real modules (candidate_list and timepoint_list) which means I am king of breaking the rules of unit testing here because I am not only testing the `getPermissionsVerbose()` function but also the `getActiveModulesIndexed()` function of the module class. I know we should instead be using a mock module but I'm not sure how to do that. let me know if and how you want this PR changed